federation/pkg/kubefed/init/init.go

```diff
@@ -583,9 +587,51 @@ func createControllerManager(clientset *client.Clientset, namespace, name, svcNa
 		},
 	}
 
+	if dnsProviderConfig != "" {
```

Comment:
Having to validate this change by exercising the whole of the controller manager init is more work than necessary.  Consider factoring these additions into separate function(s) to simplify unit testing, e.g.

```golang
func addDNSProviderConfig(dnsProviderConfig string, dep *extensions.Deployment) {
  if dnsProviderConfig != "" {
    ...
  }
}
```